Tests/Functional/Action/Platform/OidcLoginAuthenticationActionTest.php

```diff
@@ -0,0 +1,176 @@
+                'state' => 'state',
+                'response_mode' => 'form_post',
+                'nonce' => 'nonce',
+                'prompt' => 'none',
+                'lti_message_hint' => $oidcLaunchRequest->getLtiMessageHint(),
+                'lti_deploymentId' => $oidcLaunchRequest->getLtiDeploymentId(),
+            ]
+        );
+
+        $this->assertLoginAuthenticationResponse($this->client->getResponse());
+    }
+
+    public function testValidLoginAuthenticationWithGetMethod(): void
+    {
+        /** @var OidcLaunchRequest $oidcLaunchRequest */
+        $oidcLaunchRequest = static::$container
+            ->get(OidcLaunchRequestBuilder::class)
+            ->buildResourceLinkOidcLaunchRequest(
+                new ResourceLink('resourceLinkIdentifier'),
+                $this->registration,
+                'loginHint'
+            );
+
+        $this->client->request(
+            Request::METHOD_GET,
+            sprintf(
+                '/oidc/login-authentication?%s',
+                http_build_query([
+                    'scope' => 'openid',
+                    'response_type' => 'id_token',
+                    'client_id' => $this->registration->getClientId(),
+                    'redirect_uri' => $this->registration->getTool()->getLaunchUrl(),
+                    'login_hint' => 'loginHint',
+                    'state' => 'state',
+                    'response_mode' => 'form_post',
+                    'nonce' => 'nonce',
+                    'prompt' => 'none',
+                    'lti_message_hint' => $oidcLaunchRequest->getLtiMessageHint(),
+                    'lti_deploymentId' => $oidcLaunchRequest->getLtiDeploymentId(),
+                ])
+            )
+        );
+
+        $this->assertLoginAuthenticationResponse($this->client->getResponse());
+    }
+
+    public function testLoginAuthenticationWithInvalidLtiMessageHint(): void
+    {
+        $this->client->request(
+            Request::METHOD_POST,
+            '/oidc/login-authentication',
+            [
+                'scope' => 'openid',
+                'response_type' => 'id_token',
+                'client_id' => $this->registration->getClientId(),
+                'redirect_uri' => $this->registration->getTool()->getLaunchUrl(),
+                'login_hint' => 'loginHint',
+                'state' => 'state',
+                'response_mode' => 'form_post',
+                'nonce' => 'nonce',
+                'prompt' => 'none',
+                'lti_message_hint' => 'invalid',
+            ]
+        );
+
+        $response = $this->client->getResponse();
+        $this->assertInstanceOf(Response::class, $response);
+        $this->assertEquals(Response::HTTP_UNAUTHORIZED, $response->getStatusCode());
+        $this->assertStringContainsString('OIDC login authentication failed', (string)$response->getContent());
+    }
+
+    private function assertLoginAuthenticationResponse(Response $response): void
```

Comment:
If you typehint the parameter, the instanceOf assert is not necessary anymore, as it will be impossible for it to fail

Reply:
already fixed